src/adams_methods.jl

```diff
@@ -0,0 +1,626 @@
+    # for variable order calculations, which correspond to infinity norms of:
+    #      LE(n+1)_{k-1}, LE(n+1)_{k}, E(n+1)_{k+1}, and LE(n+1)_{k+2}
+    # from Hairer pgs 422-424
+    err_km1, err_k, err_kp1, err_kp2 = 0.0,0.0,0.0,0.0
+
+    #-##########################################################################
+    # Variable Step Variable Order (VSVO) Step Integration Loop
+    #-##########################################################################
+    progressmeter
+    if progressmeter
+        prog = Progress(round(Int,1000*tdir*(tfinal-tstart)), "Solving ODE:")
+    end
+    while  tfinal*tdir > t[n]*tdir
+        #--#####################################################################
+        # Attempt a step with current trial stepsize `dt`
+        #--#####################################################################
+        successful_step = false
+        while !successful_step
+            #---################################################################
+            # Make an approriate time step with current trial stepsize `dt`
+            #---################################################################
+            #Make the step
+            t_np1 = t[n]+dt
+            #---################################################################
+            # Calculating explicit approximatiion to y_np1 with current step dt
+            #---################################################################
+
+            # Calculation of g_{j}(n)coefficients for j=0,...,k+1;
+            g_coefs!(g,c,dt,t,t_np1,n,min(n,(num_stages+1)+1))
+            #@show min(max(n-1,1),(num_stages)+1)
+            # Calculation of  ϕstar_{j}(n), and ϕ_{j}(n) coef. for j=0,...,k;
+            #ϕ_slots = min(max(n-1,1),(num_stages)+1)
+            ϕ_and_ϕstar_coefs!(ϕstar_n, ϕ_n, ϕstar_nm1, b,dy_n,t,t_np1,n,1:(num_stages)+1)
+            #Note that  1<= min(max(n-1,1),num_stages+1) <= n-1 and num_stages+1
+
+            # Calculate predictor with explicit method of order `num_stages`. See
+            # equation (5.5) in Hairer et al, Vol 1, where we note that here
+            # y_np1 = y_{k}(t_{n+1})
+
+            copy!(y_np1,y_n)
+            for j= 0:num_stages-1
+                y_np1 += dt*g[(j)+1]*ϕstar_n[(j)+1]
+            end
+
+            # Compute current estimate for derivative
+            dy_np1=F(t_np1,y_np1)
+
+            #---################################################################
+            # Correction-Evaluaion (CE) Loop for implicit solver
+            #     See Hairer p. 360 for a general description of this method
+            #---################################################################
+
+            # Set CE_loops to be how many correction-evaulation cycles
+            # desired. As least one loop is necessary, for the sake of the
+            # adaptive step selection
+            CE_loops = 1
+            for loop_counter=1:CE_loops
```

Comment:
This is fine for now, but it should really have the option to loop until some tolerance between the previous and the current is satisfied.